### ISSUE TITLE
missing python module compilation

### ISSUE THREAD

**Issue by nzasch** [lofar-astron/LOFARBeam#14]
Hi

compiling LOFARBeam commit 38be4e449b84325fd47fff58c9507fc37cbf7216 with command 
```
cmake3 .. -DCMAKE_INSTALL_PREFIX=/opt/LOFARBeam-38be4e449b84325fd47fff58c9507fc37cbf7216 -DCASACORE_ROOT_DIR=/opt/casacore-2.4.1/ -DBOOST_ROOT=/opt/boost/ -DCMAKE_CXX_STANDARD=11
```

gives:
```
-- The C compiler identification is GNU 7.3.1
-- The CXX compiler identification is GNU 7.3.1
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc
-- Check for working C compiler: /opt/rh/devtoolset-7/root/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++
-- Check for working CXX compiler: /opt/rh/devtoolset-7/root/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for sgemm_
-- Looking for sgemm_ - not found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Looking for sgemm_
-- Looking for sgemm_ - found
-- Found BLAS: /opt/OpenBLAS/lib/libopenblas.so  
-- Looking for cheev_
-- Looking for cheev_ - found
-- A library with LAPACK API found.
CMake Warning (dev) at CMake/FindCasacore.cmake:135 (find_package):
  Policy CMP0074 is not set: find_package uses <PackageName>_ROOT variables.
  Run "cmake --help-policy CMP0074" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.

  Environment variable HDF5_ROOT is set to:

    /opt/hdf5

  For compatibility, CMake is ignoring the variable.
Call Stack (most recent call first):
  CMake/FindCasacore.cmake:227 (casacore_find_package)
  CMakeLists.txt:8 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- HDF5: Using hdf5 compiler wrapper to determine C configuration
-- Found HDF5: /opt/hdf5-1.8.21/lib/libhdf5.so;/usr/lib64/libz.so;/usr/lib64/libdl.so;/usr/lib64/libm.so (found version "1.8.21")  
-- Found the following Casacore components: 
--   ms: /opt/casacore-2.4.1/lib/libcasa_ms.so
--   measures: /opt/casacore-2.4.1/lib/libcasa_measures.so
--   scimath: /opt/casacore-2.4.1/lib/libcasa_scimath.so
--   tables: /opt/casacore-2.4.1/lib/libcasa_tables.so
--   scimath_f: /opt/casacore-2.4.1/lib/libcasa_scimath_f.so
--   casa: /opt/casacore-2.4.1/lib/libcasa_casa.so
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Found PythonInterp: /usr/bin/python (found version "2.7.5") 
-- Found PythonLibs: /usr/lib64/libpython2.7.so (found version "2.7.5") 
-- Build directory for Python extensions:        /opt/src/LOFARBeam-38be4e449b84325fd47fff58c9507fc37cbf7216/build/lib64/python2.7/site-packages
-- Installation directory for Python extensions: /opt/LOFARBeam-38be4e449b84325fd47fff58c9507fc37cbf7216/lib64/python2.7/site-packages
-- Boost version: 1.64.0
-- Found the following Boost libraries:
--   python
-- Found the following Casacore components: 
--   python: /opt/casacore-2.4.1/lib/libcasa_python.so
--   casa: /opt/casacore-2.4.1/lib/libcasa_casa.so
-- Configuring done
-- Generating done
-- Build files have been written to: /opt/src/LOFARBeam-38be4e449b84325fd47fff58c9507fc37cbf7216/build
```

but then giving `make && make install` 
compiles without errors but does not produce an usable python module

normally lofar release compilation produced /opt/lofar-3.2.18/lib64/python2.7/site-packages/lofar/

am I missing something?
thanks
francesco

**Comment by tammojan** [comment]
You're not missing something, nothing is installed.

**Comment by tammojan** [comment]
@basvandertol could you have a look?

**Comment by basvandertol** [comment]
Yes, made a fix, see pull request #16

**Comment by tammojan** [comment]
@nzasch Please update to the latest version (master branch) and try again.

You will need to set the `PYTHONPATH` to include the new library, in your case you'll need
```
export PYTHONPATH=/opt/LOFARBeam-38be4e449b84325fd47fff58c9507fc37cbf7216/lib/python2.7/site-packages/lofar/:$PYTHONPATH
export LD_LIBRARY_PATH=/opt/LOFARBeam-38be4e449b84325fd47fff58c9507fc37cbf7216/lib/:$LD_LIBRARY_PATH
```

Please let us know if this works.

We should be able to make setting the `LD_LIBRARY_PATH` not necessary anymore, I'll open an issue for that.

**Comment by aroffringa** [comment]
@nzasch @tammojan Is this issue solved?

**Comment by bennahugo** [comment]
Hi I still need to touch a ```__init__.py``` file to get this to import (in the absence of other lofar packages)
```
#####################################################################
## BUILD LOFAR FROM SOURCE
#####################################################################
WORKDIR /src
RUN git clone https://github.com/lofar-astron/LOFARBeam.git
WORKDIR LOFARBeam
RUN git checkout db8d082b34c203417b01d20b6f3bf0df17728ee8
#RUN svn co --non-interactive --no-auth-cache --username lofar-guest --password lofar-guest https://svn.astron.nl/LOFAR/tags/LOFAR-Release-4_0_9
#WORKDIR LOFAR-Release-4_0_9
#RUN wget https://codeload.github.com/lofar-astron/LOFARBeam/tar.gz/v4.0
#RUN tar xvf v4.0
RUN mkdir -p build/gnucxx11_opt
WORKDIR build/gnucxx11_opt
RUN cmake -DBUILD_PACKAGES="pystationresponse" -DCMAKE_INSTALL_PREFIX=/usr ../../
RUN make -j16
RUN make install
WORKDIR /
RUN touch /usr/lib/python2.7/site-packages/lofar/__init__.py
ENV PYTHONPATH "/usr/lib/python2.7/site-packages:$PYTHONPATH"
RUN python -c "import lofar.stationresponse as lsr"
```

**Comment by aroffringa** [comment]
Fixed in #18 .
